src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,149 @@
+                            taskList.get(index).markAsDone();
+                            isTaskListUpdated = true;
+
+                            ui.printMessage("Nice! I've marked this task as done:\n  " + taskList.get(index));
+                            break;
+                        case DELETE:
+                            // -1 to account for zero-indexing
+                            Task removedTask = taskList.remove(Integer.parseInt(command.getArgs()[0]) - 1);
+                            isTaskListUpdated = true;
+
+                            ui.printMessage("Noted. I've removed this task:\n  " + removedTask + "\nNow you have "
+                                    + taskList.size() + " tasks in the list.");
+                            break;
+                        case TODO:
+                            Todo newTodo = new Todo(userInput.substring(5).trim());
+                            this.taskList.add(newTodo);
+                            isTaskListUpdated = true;
+
+                            ui.printMessage("Got it. I've added this task:\n  " + newTodo + "\nNow you have " + this.taskList.size()
+                                    + " tasks in the list.");
+                            break;
+                        case DEADLINE:
+                            description = command.getArgs()[0];
+                            dateTime = LocalDate.parse(command.getArgs()[1]);
+                            Deadline newDeadline = new Deadline(description, dateTime);
+                            taskList.add(newDeadline);
+                            isTaskListUpdated = true;
+
+                            ui.printMessage("Got it. I've added this task:\n  " + newDeadline + "\nNow you have "
+                                    + taskList.size() + " tasks in the list.");
+                            break;
+                        case EVENT:
+                            description = command.getArgs()[0];
+                            dateTime = LocalDate.parse(command.getArgs()[1]);
+                            Event newEvent = new Event(description, dateTime);
+                            taskList.add(newEvent);
+                            isTaskListUpdated = true;
+
+                            ui.printMessage("Got it. I've added this task:\n  " + newEvent + "\nNow you have " + taskList.size()
+                                    + " tasks in the list.");
+                            break;
+                        case DATE:
+                            LocalDate queryDate = LocalDate.parse(userInput.substring(5));
+                            TaskList dueTasks = taskList.filterByDate(queryDate);
+
+                            ui.printTaskList(dueTasks, queryDate);
+                            break;
+                        case FIND:
+                            // -1 to account for zero-indexing
+                            keyword = command.getArgs()[0];
```

Comment:
If I'm not wrong, the indentation of the case should be 4 spaces to the left. Please do check as I may be wrong!